src/cogent3/core/new_sequence.py

```diff
@@ -2298,7 +2342,7 @@ def __init__(
         if step == 0:
             raise ValueError("step cannot be 0")
         step = 1 if step is None else step
-
+        self.alphabet = alphabet
```

Comment:
as discussed, adding `__slots__` to all ABC's ensures classes that inherit from them and use slots will behave as expected.